Add tests for "Leave meeting" Functionality

pinhaum opened the pull request:
### What does this PR do?
This PR introduces a new test for the "leave meeting" functionality in BigBlueButton.

#### Changes

- Added a test to cover user leave meeting behavior.

#### Test steps:

1. Moderator and user sessions are initialized.
2. The moderator leaves the meeting.
3. The test checks that the moderator left the meeting and the UI shows the meeting ended modal.
4. reproduces steps 2. and 3. for the atendee

#### Video examples of the tests running:

- for moderator

[for moderator](https://github.com/user-attachments/assets/32653411-4845-4259-885a-aa53726e8911)


- for user

[for user](https://github.com/user-attachments/assets/c0c0b250-04c5-4368-af90-220df5103a4d)


### Motivation
Automating the "Leave meeting" scenario helps guarantee that this workflow operates as expected.


### How to test

1. run playwright user tests with: `npm run test user` or `npx playwright test user`
    - you can also run the specific test using the command: `npm run test:filter "Leave Meeting"`
2. Confirm that the end meeting test passes and UI behaves as expected.

antobinary committed:
chore: Bump bbb-playback to 5.3.5

https://github.com/bigbluebutton/bbb-playback/releases/tag/v5.4.0

pinhaum committed:
test: add data-test attribute to confirmModal for end meeting

paultrudel committed:
Add pres upload concurrency explanation

prlanzarin committed:
fix(livekit): unable to connect if /stuns maxTtl is undefined

If maxTtl is not defined for the configured ICE servers returned by
the /stuns endpoint, LiveKit won't be able to connect. The hook used to
fetch said ICE servers gets stuck in a loading loop due to the cache
being perpetually invalidated because of the nil TTL.

Assume that if no TTL is set for /stuns, the cache is valid
indefinitely.

Tainan404 committed:
Fix(html5): fix bad state caused by deduplicated subscription, inspired by PR #23657

Tainan404 committed:
Fix(html5): Error on missing prop on button component

pinhaum committed:
test: add leave meeting test

pinhaum committed:
refac: remove commented section

Tainan404 committed:
Fix(html5): required props returning undefined

JoVictorNunes committed:
docs: Improve userdata-bbb_initial_selected_tool docs

ramonlsouza committed:
fix: Listen-only participants show misleading crossed-out microphone icon instead of headphones icon (#23672)

* track listen only state with transparentListenOnly

* fix incorrect values and types

* Update akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/voice/VoiceApp.scala

Co-authored-by: Gustavo Trott <gustavo@trott.com.br>

* remove unnecessary check

* remove userId from body of userSetListenOnlyInput action

* remove unused code

* Update bbb-common-message/src/main/scala/org/bigbluebutton/common2/msgs/VoiceConfMsgs.scala

Co-authored-by: Gustavo Trott <gustavo@trott.com.br>

---------

Co-authored-by: Gustavo Trott <gustavo@trott.com.br>

ramonlsouza committed:
fix: Listen-only participants show misleading crossed-out microphone icon instead of headphones icon (#23672)

* track listen only state with transparentListenOnly

* fix incorrect values and types

* Update akka-bbb-apps/src/main/scala/org/bigbluebutton/core/apps/voice/VoiceApp.scala

Co-authored-by: Gustavo Trott <gustavo@trott.com.br>

* remove unnecessary check

* remove userId from body of userSetListenOnlyInput action

* remove unused code

* Update bbb-common-message/src/main/scala/org/bigbluebutton/common2/msgs/VoiceConfMsgs.scala

Co-authored-by: Gustavo Trott <gustavo@trott.com.br>

---------

Co-authored-by: Gustavo Trott <gustavo@trott.com.br>

JoVictorNunes committed:
fix(bbb-export-annotations): Text wrapping in text shapes

JoVictorNunes committed:
fix(bbb-export-annotations): Add proper validations

antonbsa committed:
chore: update bigbluebutton-html-plugin-sdk to version 0.0.90 (#23686)

GuiLeme committed:
feat(html5): ui-command to better manipulate generic content in sidekick area (#23606)

* [sidekick-area-commands] added 4 new ui- commands for sidekick-area

* [sidekick-area-commands] synced with sdk

* [sidekick-area-commands] refactor to reduce number of states in the class

* [sidekick-area-commands] added callback to optimize rendering

prlanzarin committed:
build(bbb-webrtc-recorder): v0.11.0

v0.11.0
---
* feat(livekit): generation of rtpdump files for recorded tracks
* fix(mediasoup): stop event triggered by flowing=false if adapter is closed

prlanzarin committed:
build(bbb-webrtc-sfu): v2.19.0-beta.5

v2.19.0-beta.5
---
* feat(livekit): add flag to enable custom BBB permission checks
* feat(livekit): add metric for track publish handler errors
* fix: bbb-webrtc-recorder stop requests have no timeout
* fix: camera and screen states in BBB become stale on broken stop calls

gustavotrott committed:
Docs (plugins): Add example of javascriptEntrypointIntegrity and remove unimplemented features (#23692)

* docs plugins: add example of javascriptEntrypointIntegrity and remove unimplemented features

antobinary committed:
Update bbb-pads.placeholder.sh

transifex-integration[bot] committed:
Translate en.json in uk_UA (#23689)

100% translated source file: 'en.json'
on 'uk_UA'.

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

antobinary committed:
chore: Bump release to 3.0.13

gabriellpr committed:
updating smart slides test

ramonlsouza committed:
adjust locale dropdown to not display fallback id + include ru-RU en string

ramonlsouza committed:
fix: Poll Results in the pdf-File (#23694)

* generate poll results svg

* adjustments

* remove unused code

ramonlsouza committed:
fix fullscreen shortcut not working inside the whiteboard

JoVictorNunes committed:
improvement(bbb-learning-dashboard): Avoid function reinstantiations on every render

GuiLeme committed:
[tweak-floating-window] Adjust z-index and window boundaries

JoVictorNunes committed:
fix(bbb-learning-dashboard): Wrong poll answer count

JoVictorNunes committed:
fix(bbb-learning-dashboard): Quizzes table sorting toggling doesn't work

JoVictorNunes committed:
Remove wrapping array

hiroshisuga committed:
Update component.jsx

hiroshisuga committed:
Update component.jsx

GuiLeme committed:
[tweak-floating-window] fix issue where floating window removes camera-dock from sight

antonbsa reviewed (commented):
test works fine, but I'd suggest some changes in the steps in order to track a few more expected behaviors:

- join 2 users
- 1 user leaves meeting (check `meetingEndedModal` and `redirectButton`)
- other user waits for this user's item in the user list to be removed (and maybe also the user count in "Users (1)" to decrease?)

what you think?

Tainan404 committed:
Feat(html5): Add hot reload environment variable

ramonlsouza committed:
fix quick external video position when presentation is fullscreen

gustavotrott committed:
dev bbb-web: add parameter --build to re-build bbb-common-web

gustavotrott committed:
dev html5 deploy.sh: avoid building legacy safari build every time

gustavotrott committed:
tweak node scripts to automatically identify when it needs to run npm install

ramonlsouza committed:
fix leave meeting button styles

ramonlsouza committed:
adjust dropdown component right click to only happen if the dropdown is open

transifex-integration[bot] committed:
Updates for project BigBlueButton v3.0 HTML5 client and language nb_NO (#23734)

* Translate en.json in nb_NO

100% translated source file: 'en.json'
on 'nb_NO'.

* Translate en.json in nb_NO

100% translated source file: 'en.json'
on 'nb_NO'.

* Translate en.json in nb_NO

100% translated source file: 'en.json'
on 'nb_NO'.

---------

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

transifex-integration[bot] committed:
Updates for project BigBlueButton v3.0 HTML5 client and language ja (#23727)

* Translate en.json in ja

100% translated source file: 'en.json'
on 'ja'.

* Translate en.json in ja

100% translated source file: 'en.json'
on 'ja'.

---------

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

gustavotrott committed:
Add new graphql flag componentFlags.hasSharedNotes

gustavotrott committed:
Add new graphql flag componentFlags.hasSharedNotes

hiroshisuga committed:
[3.0 Quick poll] number of options up to 19 (#23698)

* extend poll option

* Update component.jsx

Additiona modification.

* remove an unintentional space

Tainan404 committed:
Fix(html5): Video preview modal keeps frozen when using bbb_auto_share_webcam and bbb_skip_video_preview

transifex-integration[bot] committed:
Translate en.json in it_IT (#23741)

100% translated source file: 'en.json'
on 'it_IT'.

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

Tainan404 committed:
Ref(html5): Fetch IsBreakout information from useMeeting and removes getIsBreakout subscription

gustavotrott committed:
Introduces new endpoint /api/rest/meetingStaticData to fetch all static data at the beginning of the meeting

JoVictorNunes committed:
fix(html5/whiteboard): Do not run resync code on change page
- Do not run shape resync code every time the current page changes, as it is intended to run on reconnect.

Tainan404 committed:
Ref(html5): Fetch isBreakout information from useMeeting and removes MeetingIsBreakout subscription

Tainan404 committed:
Ref(html5): Add skip condition to hasPendingPoll subscription, only running when having a active poll

Tainan404 committed:
Fix(bbb-graphql-server): Typo on componentsFlags, isSharedNotedPinned to isSharedNotesPinned

Tainan404 committed:
Ref(html5): Fetch isSharedNotesPinned  from useMeeting and removes isSharedNotesPinned query

Tainan404 committed:
Ref(html5): Removes PollPublished subscription, presentation is restored bacause poll is an annotation

Tainan404 committed:
Ref(html5): Unify EXPORTING_PRESENTATIONS_SUBSCRIPTION, PRESENTATIONS_SUBSCRIPTION and PRESENTATIONS_SUBSCRIPTION subscriptions

Tainan404 committed:
Ref(html5): Refactor useScreenshare hook to use skip and only execute when have an active screenshare

Tainan404 committed:
Ref(html5): Make getCaptions only runs when it's enable and have a valide locale

Tainan404 committed:
Ref(html5): Improve pads logic to render padContent only for users with no permission to render shared notes

Tainan404 committed:
Ref(html5): Removes hasPad subscription in favor of hasSharedNotes flag

Tainan404 committed:
Fix(html5): Includes hasCameraAsContent flag as condition for screnshare subscription execution

GuiLeme committed:
ref: Added information to useUsersBasicInfo and ordered the results (#23688)

* [order-users] refactored users-basic-info to be ordered and include more information

* [order-users] link SDK pr

* [order-users] remove some properties from users-basic-info

* [order-users] bump version of SDK

transifex-integration[bot] committed:
Translate en.json in el_GR (#23752)

100% translated source file: 'en.json'
on 'el_GR'.

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

transifex-integration[bot] committed:
Translate en.json in ja (#23751)

100% translated source file: 'en.json'
on 'ja'.

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

transifex-integration[bot] committed:
Translate en.json in eu (#23746)

100% translated source file: 'en.json'
on 'eu'.

Co-authored-by: transifex-integration[bot] <43880903+transifex-integration[bot]@users.noreply.github.com>

prlanzarin committed:
build(bbb-webrtc-sfu): v2.19.0-beta.6

v2.19.0-beta.6
---
* feat(livekit): DTMF handling
* feat(livekit): implement SIP speaking detection
* fix(livekit): incorrect parsing of dispatch and trunk rule options
* fix(livekit): SIP users have hardcoded `Phone ` prefix in their names
* fix(livekit): unhandled EjectUserFromVoiceConfSysMsg
* build: @livekit/rtc-node@0.13.18
* build: transitive dep brace-expansion audit fix

Tainan404 committed:
Ref(html5): Make useMeeting use MeetingStaticData

Tainan404 committed:
Fix(graphql-server): recordingPolicies returning null on api

Tainan404 committed:
Fix(html5): TS compilation

pinhaum commented:
Closed in favor of #23811
